src/app/admin/add-category/add-category.page.ts

```diff
@@ -1,15 +1,133 @@
+import { FormGroup, FormControl, Validators } from '@angular/forms';
 import { Component, OnInit } from '@angular/core';
+import { CategoriesService } from '../../../providers/services/categories/categories.service';
+import { CONSTANTS } from '../../../providers/utils/constants';
+import { WidgetUtilService } from '../../../providers/utils/widget';
+import { DashboardService } from '../../../providers/services/dashboard/dashboard.service';
+import { ProfileModel } from '../../../providers/models/profile.model';
+import { StorageServiceProvider } from '../../../providers/services/storage/storage.service';
+
 
 @Component({
   selector: 'app-add-category',
   templateUrl: './add-category.page.html',
   styleUrls: ['./add-category.page.scss'],
+  providers: [CategoriesService, DashboardService]
 })
 export class AddCategoryPage implements OnInit {
+  skipValue = 0;
+  limitValue = CONSTANTS.PAGINATION_LIMIT;
+  selectedCategoryType: string = 'parent';
+  categoryTypeList = ['parent', 'child']
+  categoryList: Array<any>;
```

Comment:
Define a model for category list response and assign a type to list in every place where you fetch categories.